modules/install-consul/install-consul

```diff
@@ -166,8 +166,46 @@ function fetch_binary {
   local -r version="$1"
   local download_url="$2"
 
-  if [[ -z "$download_url" && -n "$version" ]];  then
-    download_url="https://releases.hashicorp.com/consul/${version}/consul_${version}_linux_amd64.zip"
+  local -r cpu_arch="$(uname -m)"
+  local binary_arch=""
+  case "$cpu_arch" in
+    x86_64)
+      binary_arch="amd64"
+      ;;
+    x86)
+      binary_arch="386"
+      ;;
+    arm64|aarch64)
+      binary_arch="arm64"
+      ;;
+    arm*)
+      # The following info is taken from https://www.consul.io/downloads
+      #
+      # Note for ARM users:
+      #
+      # Use Armelv5 for all 32-bit armel systems
+      # Use Armhfv6 for all armhf systems with v6+ architecture
+      # Use Arm64 for all v8 64-bit architectures
+      # The following commands can help determine the right version for your system:
+      #
+      # $ uname -m
+      # $ readelf -a /proc/self/exe | grep -q -c Tag_ABI_VFP_args && echo "armhf" || echo "armel"
+      #
+      local -r vfp_tag="$(readelf -a /proc/self/exe | grep -q -c Tag_ABI_VFP_args)"
+      if [[ ! $vfp_tag  ]]; then
```

Comment:
Are you checking for an empty string here? If so, `-z` is the typical way to do that.

Reply:
Yep, applied the suggestion and used `-z` for checking empty string. 👍